WIP: BT FR14 - Appliances that have questions during the deploy I miss that option in the portal

nvazquez opened the pull request:
In progress PR

nvazquez committed:
Test file created

nvazquez committed:
Add OVF properties parser and retriever

nvazquez committed:
Pass OVF properties when available as part of download answer and persist as details

nvazquez committed:
Fix persist OVF properties

nvazquez committed:
Remove test file created previously

nvazquez committed:
Remove unnecessary property

nvazquez committed:
Store OVF vApp properties on separate table

nvazquez committed:
Fix persisting OVF properties

nvazquez committed:
Add removed field

nvazquez committed:
In progress - Display OVF properties on the UI

yadvr committed:
ui: don't ignore project when listing templates (#3233)

Problem: When uploading template or iso in a project, the progress is not seen
Root Cause: When users upload a template/iso in project view, the template/iso is not visible in the all filter. This creates confusion that template/iso has not been uploaded.
Solution: Since the api listtemplates with a projectid acts is a superset of list templates without a project id, we should not ignore the project id when listing templates in all filter.

nvazquez committed:
Add OVF properties as part of listTemplates response

yadvr commented:
can you close this and open a WIP PR on upstream @nvazquez ? (if not done already?)

yadvr committed:
storage: publish delete usage event for snapshot deletion (#3212)

Problem: Users are billed for destroyed VMs with VM snapshots because usage records don't get that the VM and VM snapshots are removed.
Root Cause: The destroyVirtualMachine and expungeVirtualMachine APIs were removing VM snapshots but not generating VMSNAPSHOT.DELETE usage event due to which the VM snapshots were not marked as removed in the usage_vmsnapshot table.
Solution: The issue was fixed by emitting the proper usage event for all the VM snapshots of a VM that is destroyed.

shwstppr committed:
ui: show complete domain for accounts (#2994) (#3256)

This allows showing complete domain, ie, domain path for accounts list view and account detail.
Added a new key, domainpath, in AccountResponse.

Fixes #2994

Signed-off-by: Abhishek Kumar <abhishek.mrt22@gmail.com>

nvazquez committed:
Add new tab for OVF properties, skipped if not available

nvazquez committed:
Display new tab for OVF properties

nvazquez committed:
Pass OVF properties if available to deployVM cmd

nvazquez committed:
Refactor template OVF properties dao and persist VM OVF properties

nvazquez commented:
Sure @rhtyd - using [https://github.com/apache/cloudstack/pull/3271/](https://github.com/apache/cloudstack/pull/3271/)
